src/encoded/viewconfigs/matrix.py

```diff
@@ -202,50 +196,60 @@ def preprocess_view(self):
                 type_info = self._types[self._doc_types[0]]
                 self._schema = type_info.schema
         self._validate_items(type_info)
-        self._result['title'] = self._set_result_title(type_info)
-        self._result['matrix'] = type_info.factory.matrix.copy()
-        self._matrix = self._result['matrix']
-        self._matrix['x']['limit'] = self._request.params.get('x.limit', 20)
-        self._matrix['y']['limit'] = self._request.params.get('y.limit', 5)
+        matrix = type_info.factory.matrix.copy()
+        matrix['x']['limit'] = self._request.params.get('x.limit', 20)
+        matrix['y']['limit'] = self._request.params.get('y.limit', 5)
         search_route = self._request.route_path('search', slash='/')
-        self._matrix['search_base'] = search_route + self._search_base
-        matrix_route = self._request.route_path('matrix', slash='/')
-        self._matrix['clear_matrix'] = matrix_route + '?type=' + self._doc_types[0]
-        self._result['views'] = self._construct_result_views(type_info)
-        query, used_filters = self._construct_query()
-        es_results = self._elastic_search.search(body=query, index=self._es_index)
+        matrix['search_base'] = search_route + self._search_base
+        matrix['clear_matrix'] = matrix_route + '?type=' + self._doc_types[0]
+        search_result = {
+            '@context': self._request.route_path('jsonld_context'),
+            'filters': [],
+            '@id': matrix_route + self._search_base,
+            '@type': ['Matrix'],
+            'notification': '',
+            'title': self._set_result_title(type_info),
+            'matrix': matrix,
+            'view': self._construct_result_views(type_info),
+        }
+        search_query, used_filters = self._construct_query(search_result, matrix)
+        es_results = self._elastic_search.search(
+            body=search_query,
+            index=self._es_index
+        )
         aggregations = es_results['aggregations']
         total = aggregations['matrix']['doc_count']
-        self._result['matrix']['doc_count'] = total
-        self._result['matrix']['max_cell_doc_count'] = 0
-        self._result['facets'] = self._format_facets(
+        search_result['matrix']['doc_count'] = total
+        search_result['matrix']['max_cell_doc_count'] = 0
+        search_result['facets'] = self._format_facets(
```

Comment:
Don't add the facets to the search_results since it won't exist at this point due to other change.

Reply:
Let me know if this issue still exist. I refactored but I could not see a reason to move this.